src/main/java/org/wildfly/security/ssl/CipherSuiteSelector.java

```diff
@@ -540,9 +540,14 @@ private static CipherSuitePredicate getSimplePredicateByName(final String word)
             case "AES256":      return CipherSuitePredicate.matchEncryption(Encryption.AES256, Encryption.AES256GCM);
             case "AES":         return CipherSuitePredicate.matchEncryption(Encryption.AES128, Encryption.AES128GCM, Encryption.AES256, Encryption.AES256GCM);
             case "AESGCM":      return CipherSuitePredicate.matchEncryption(Encryption.AES128GCM, Encryption.AES256GCM);
+            case "ARIA":        return CipherSuitePredicate.matchEncryption(Encryption.ARIA128, Encryption.ARIA128GCM, Encryption.ARIA256, Encryption.ARIA256GCM);
+            case "ARIA128":     return CipherSuitePredicate.matchEncryption(Encryption.ARIA128, Encryption.ARIA128GCM);
+            case "ARIA256":     return CipherSuitePredicate.matchEncryption(Encryption.ARIA256, Encryption.ARIA256GCM);
             case "CAMELLIA128": return CipherSuitePredicate.matchEncryption(Encryption.CAMELLIA128);
             case "CAMELLIA256": return CipherSuitePredicate.matchEncryption(Encryption.CAMELLIA256);
             case "CAMELLIA":    return CipherSuitePredicate.matchEncryption(Encryption.CAMELLIA128, Encryption.CAMELLIA256);
+            case "CHACHA20-POLY1305":
+                                return CipherSuitePredicate.matchEncryption(Encryption.CHACHA20POLY1305);
```

Comment:
On further reading, I think this should be two predicates.  One, "CHACHA20", is a cipher and would go under `Encryption`.  The other, "POLY1305", is a MAC.  If we want it as a predicate (OpenSSL only defines a predicate for "CHACHA20"), I think it would belong as a `Digest`.  RFC 7905 describes it as an "authenticator", but it appears they're talking about message integrity, not certificate authentication, which makes sense as the combination of POLY1305 and CHACHA20 is a stream cipher with integrity support.

When I said OpenSSL used "CHACHA20-POLY1305", I was looking in the actual cipher strings, which was my mistake.

Note that the mechanism database would also need fixing since it refers to the encryption as "CHACHA20POLY1305" and digest as "AEAD", whereas I would think we'd want the encryption to be known as "CHACHA20" and the digest as "POLY1305", which would allow the predicate(s) to work correctly.  Another supporting reason for this is that, according to Wikipedia, historically there was a AES/Poly1305 combination as well as a Salsa20/Poly1305 variation (Salsa20 is a predecessor of ChaCha20) used in various areas.  In addition, it is probably possible to use ChaCha20 with some other MAC.

Reply:
Yeah, I was a bit confused by that, now it makes sense. Well, I think we can go with your PR and I will create additional jira+commit for the ARIA ciphers.